src/serena/agent.py

```diff
@@ -47,8 +47,12 @@ def __init__(self, config_dict: dict[str, Any], project_name: str, project_root:
         if project_root is None:
             project_root = Path(config_dict["project_root"])
         self.project_root: str = str(project_root.resolve())
-        self.ignored_dirs: list[str] = config_dict.get("ignored_dirs", [])
+        self.ignored_paths: list[str] = config_dict.get("ignored_paths", [])
         self.excluded_tools: set[str] = set(config_dict.get("excluded_tools", []))
+        self.ignore_all_files_in_gitignore = config_dict.get("ignore_all_files_in_gitignore", True)
```

Comment:
Same here. We should not assume that a default is OK.

Reply:
I see that you (perhaps unexpectedly) handled backward compatbility below.

Reply:
If a default is OK, it is False instead of True, because that's what users had previously.

Reply:
I would rather roll it out automatically to those who already setup their configs. It's not like they consciously chose to not include gitignore, and I suspect it's a behavior that most if not all users would want

Reply:
The option you chose can cause unexpected behaviour.
The two valid options are:
* False as a default (unchanged behaviour)
* forcing a decision via an exception

Reply:
I agree, let's force. I added a callout section to the readme warning about this, and to the changelog about the breaking changes in the config